cli/src/semgrep/lsp/convert.py

```diff
@@ -11,50 +25,147 @@
 
 
 def findings_severity_to_diagnostic(severity: RuleSeverity) -> DiagnosticSeverity:
+    """Convert a rule severity to a LSP diagnostic severity"""
     return SeverityMapping.get(severity, DiagnosticSeverity.Information)
 
 
-# TODO: need to use RuleMatch here instead of raw json/dict access
-def findings_to_diagnostic(finding: JsonObject, content: str) -> JsonObject:
-    from semgrep.state import get_state
-
-    env = get_state().env
-    start_line = finding["start"]["line"] - 1
-    start_col = finding["start"]["col"] - 1
-    end_line = finding["end"]["line"] - 1
-    end_col = finding["end"]["col"] - 1
-    lines = content.split("\n")
-    if start_line == end_line:
-        match_source = lines[start_line][start_col:end_col]
-    else:
-        middle_lines = "\n".join(lines[start_line + 1 : end_line])
-        match_source = (
-            lines[start_line][start_col:] + middle_lines + lines[end_line][:end_col]
+def metavar_to_inlay(metavar: str, info: MetavarValue) -> JsonObject:
+    """Convert a Semgrep metavar to an inlay hint"""
```

Comment:
Can you add pointer to documentation on inlay hint format?

Reply:
Yea I'll add it everywhere